ui/src/app/settings/components/certs-list/certs-list.tsx

```diff
@@ -0,0 +1,253 @@
+                          getApi={(api) => this.formApiSSH = api}
+                          preSubmit={(params: NewSSHKnownHostParams) => ({
+                              data: btoa(params.data),
+                          })}
+                          validateError={(params: NewSSHKnownHostParams) => ({
+                              data: !params.data && 'SSH known hosts data is required',
+                          })}>
+                        {(formApiSSH) => (
+                            <form onSubmit={formApiSSH.submitForm} role='form' className='certs-list width-control' encType='multipart/form-data'>
+                                <div className='argo-form-row'>
+                                    <FormField formApi={formApiSSH} label='SSH known hosts data' field='data' component={TextArea}/>
+                                </div>
+                           </form>
+                        )}
+                    </Form>
+                </SlidingPanel>
+
+            </Page>
+        );
+    }
+
+    private async addTLSCertificate(params: NewTLSCertParams) {
+        try {
+            await services.certs.create({items: [{servername: params.servername, type: 'https', certdata: (params.certdata), cipher: '', certinfo: ''}], metadata: null});
+            this.showAddTLSCertificate = false;
+            this.loader.reload();
+        } catch (e) {
+            this.appContext.apis.notifications.show({
+                content: <ErrorNotification title='Unable to add TLS certificate' e={e}/>,
+                type: NotificationType.Error,
+            });
+        }
+    }
+
+    private async addSSHKnownHosts(params: NewSSHKnownHostParams) {
+        try {
+            let knownHostEntries: models.RepoCert[] = [];
+            atob(params.data).split('\n').forEach(function processEntry(item, index) {
+                const trimmedLine = item.trimLeft();
+                if (trimmedLine.startsWith('#') === false) {
+                    const knownHosts =  trimmedLine.split(' ', 3);
+                    if (knownHosts.length === 3) {
+                        // Perform a little sanity check on the data - server
+                        // checks too, but let's not send it invalid data in
+                        // the first place.
+                        const subType = knownHosts[1].match(/^(ssh\-[a-z0-9]+|ecdsa-[a-z0-9\-]+)$/ig);
+                        if (subType != null) {
+                            knownHostEntries = knownHostEntries.concat({
+                                servername: knownHosts[0],
+                                type: 'ssh',
+                                cipher: knownHosts[1],
+                                certdata: btoa(knownHosts[2]),
+                                certinfo: '',
+                            });
+                        }
+                    }
+                }
+            });
+            await services.certs.create({items: knownHostEntries, metadata: null});
```

Comment:
Please throw error if entered data is invalid:

```
if (knownHostEntries.length === 0) {
  throw new Error('No valid known hosts data entered');
}
```

Reply:
Good catch, fixed.